### PR TITLE
[LS] Refactor default value handling of optional types

### LINKED ISSUE #31790: Refactor default value handling of optional types
**Description:**

Modify `CommonUtil.getDefaultValueForType()` method to return an `Optional<String>` where it will return `Optional.empty()` when the type is an optional one.

Also add a new API to return a default value regardless of the type being an optional type.

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by TharushiJay**
## Purpose

Fixes #31790

## Approach
Modify `CommonUtil.getDefaultValueForType()` method to return an `Optional<String>`.

## Samples

As a result, it adds no return statement when generating a function with `Implement method` code action where the default return value is `()`. But for record fields where the default value is `()`, it adds `fieldName : ()`.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by mohanvive — APPROVED** [review]

**Comment by mohanvive** [comment]
@imesh waiting on your review. Can you please check and merge.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java [hunk 1]
```diff
@@ -303,15 +304,15 @@ private static String getDefaultValueForType(TypeSymbol bType, int depth) {
                 break;
             case OBJECT:
                 if (depth > MAX_DEPTH) {
-                    return "";
+                    return Optional.of("");
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java:307]: I think we should send `new ()` in this case as we have done for record type, WDYT?

**Review by malinthar — COMMENTED** [review]
Changes looks good. Have a minor comment

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java (hunk 1 above)
Comment (TharushiJay) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java:307]: In record types we return a `{}` to denote a complex record beyond level 1 and it's inserted after `Fieldname : ` (line 297). Hence it is necessary. But in the case of an object, `new()` is separately added in line 317/ 319 to the empty string returned from here. Returning `new ()` here explicitly will result in `new(new())`. Hence it is not necessary.

Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java:307]: Makes sense. Shall we do it? @TharushiJay

**Review by IMS94 — APPROVED** [review]
LGTM! Let's address @malinthar 's comment and merge this.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java (hunk 1 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java:307]: Sorry, missed @TharushiJay's reply. If that's the case, let's let this be as it is.

**Pull request merged**